src/overlay/test/OverlayTests.cpp

```diff
@@ -308,8 +347,12 @@ TEST_CASE("flow control byte capacity", "[overlay][flowcontrol]")
                     conn.getInitiator()->sendMessage(
                         std::make_shared<StellarMessage>(tx2));
 
-                    upgradeApp(app1, txSize / 2);
-                    upgradeApp(app2, txSize / 2);
+                    // We already verified that txSize2 >
+                    // MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES
+                    upgradeApp(app1,
+                               MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES);
+                    upgradeApp(app2,
+                               MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES);
```

Comment:
I think we should resize tx1 as well: the idea in this section is that we decrease max soroban tx size through an upgrade, but still allow tx1 to go through because we pick the limit as max(classic_tx_limit, soroban_tx_limit). So tx1 should be greater than MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES but within the classic limit (whereas tx2 is greater than both limits, and therefore, can't go through)

Reply:
I resized tx1 and pushed, but after reading your comment, I don't think I did this correctly. Looking over it now.

Reply:
So I think the limit decrease should be to `txSize` instead of `MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES`, but the test still passes when I decrease to `MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES`. How is tx1 able to go through?

Reply:
tx1 can go through either way: if soroban limit is too small, it still works because classic limit is enough (it's set to tx1 size in this test). if soroban limit is large (like the one you set) it doesn't really matter, since both limits are high enough for tx1. The point of this section is to verify that overlay correctly allows tx1 to go through despite soroban limit being too low (since it picks classic limit instead). So I think you need to resize tx1 at the beginning of the test to be greater than MinimumSorobanNetworkConfig::TX_MAX_SIZE_BYTES but less than tx2, then the rest of the test should work as-is.

Reply:
Ah I see. Then the test should be good now.

Reply:
thanks! looks good to me